### DISCUSSION TITLE
PRs autoclose when the branch name changes

### DISCUSSION THREAD

**Discussion by ezloj** [renovatebot/renovate#15302] [Request Help] [answered]
Was something done in respect to https://github.com/renovatebot/config-help/issues/632?
Renovate is autoclosing PRs for my repositores as well - in a situation where I wouldn't expect it to.

I'm usually running this config:

```
  fetchReleaseNotes: false,
  prConcurrentLimit: 1,
  patch: {
    groupName: 'All patches',
    description: 'Group all patch updates in a single branch/PR'
  },
  packageRules: [
    {
      matchDatasources: ['github-releases', 'github-tags'],
      enabled: false,
      description: 'Do not try to get release notes from GitHub'
    },
  ]
  ```
  
  But now I ran:
  
  ```
  "packageRules": [
        {
            "matchPackagePatterns": ["*"],
            "enabled": false
        },
        {
            "enabledManagers": ["dockerfile", "docker-compose"],
            "matchPackagePatterns": ["dbi__java17"],
            "enabled": true,
            "groupName": "CVE-2022-21449"
        },
        {
            matchDatasources: ['github-releases', 'github-tags'],
            enabled: false,
            description: 'Do not try to get release notes from GitHub'
        },
        {
            "matchManagers": ["dockerfile", "docker-compose"],
            "matchPackagePatterns": ["dbi__java17"],
            "versioning": "regex:^\\d+\\.(?<major>\\d+)\\.(?<minor>\\d+)-.*-(?<patch>\\d+)$",
            "description": "regex for java17 DBI, f.e. dbi__java17--jdk:17.0.3-7.4.3-3",
            enabled: true,
            "groupName": "CVE-2022-21449",
            "separateMajorMinor": false
        },
        {
            matchDatasources: ['docker'],
            matchUpdateTypes: [
            'major',
            'minor',
            'patch',
            ],
            enabled: true
        },
    ],
    "prConcurrentLimit": 5,
    ```
    
    And renovate is autoclosing all the PRs that were created with the first config. Why? How do I prevent that?

**Comment by viceice** [comment]
1. your release notes rule doesn't work as expected. it disables datasource (update lookup) but not release notes fetching. use a host rule to disable those.

2. changing group name causes branch name change cause new PR

**Reply by ezloj** [reply]
@viceice the config is different and the dependency (package name) is different. Does that still mean that   "prConcurrentLimit": 5, will not work? I expected   "prConcurrentLimit": 5, to work in a way that will preserve any other PRs created by renovate, regardless of other configs. As it is "top level config" - being highest - it should be decisive on the number of open PRs?

**Reply by viceice** [reply]
`prConcurrentLimit` shouldn't auto close PR, but I'm not 100% sure 🤷‍♂️

**Reply by rarkins** [reply]
No shouldn't

**Comment by ezloj** [comment]
Also, would adding an `additionalBranchPrefix` prevent the PRs being autoclosed in this case?

**Reply by viceice** [reply]
no, changing group name always cause branch name change, so new pr

**Reply by ezloj** [reply]
So in a scenario with this config and a single same repository:

```
  fetchReleaseNotes: false,
  prConcurrentLimit: 1,
  patch: {
    groupName: 'All patches',
  },
  minor {
    groupName: 'All minor changes',
  },
```

I run renovate once and it creates a pull request with group name `All minor changes`.

Then I change the config:

```
  fetchReleaseNotes: false,
  prConcurrentLimit: 1,
  patch: {
    groupName: 'All patches',
  },
```
And I guess in that case previously opened `All minor changes` PR will get autoclosed?

This prevents one use-case in renovate: when you have a base config for regular updates but in case of emergency (e.g. vulnerability patches) use a different config. In my case, I couldn't keep the same groups as I wanted to have the latest version everywhere - regardless if it's a patch, minor or major update. So I simply had to group all the changes into one. Which I guess is the reason for PRs autoclosing?

This could be solved with several renovate options:
"Fire and forget": create a pull request no matter what for the current configuration. Never autoclose created pull request and never update it (a special mark, for example in branch name?). If a pull request with the same effect (same dependencies, branch name and everything else) already exists - just do nothing (don't close, don't update, don't report error)

**Reply by rarkins** [reply]
Changing group name changes the branch name. When a branch name changes, the old PR gets closed. Either accept this or ensure the same groupSlug for both so that branch name remains the same.

**Comment by rarkins** [comment, marked as answer]
Changing branch name always causes PRs to get autoclosed unless pruneStaleBranches=false

**Comment by ezloj** [comment]
What if there's a new option in renovate:
"fire and forget": PR is created but marked as "Do not autoclose". It is never again touched or closed by renovate. PR is always created unless a PR exists that has exactly the same changes as the one that would be created. PR is created ignoring any limits or rules. 

Such an option would be helpful for "one-off" emergency updates. For example when a vulnerability is detected (such as in my case) and needs quick patching. In that case, you might have different groups (as you are likely to group patch+minor+major to always get the latest dependency version no matter what). And since you make a group - it might result in autoclosing PRs that already exist. Which isn't good in this case.

**Reply by rarkins** [reply]
I don't think that's a good fit for the way Renovate works. If you're messing around with temporary branches then I recommend you set [pruneStaleBranches](https://docs.renovatebot.com/configuration-options/#prunestalebranches) to false

**Reply by ezloj** [reply]
Fair enough. As usual, thank you for very quick help and answers.
